xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -251,23 +252,38 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
     }
   }
 
-  private static final class ServerSdsProtocolNegotiator implements ProtocolNegotiator {
+  @VisibleForTesting
+  public static final class ServerSdsProtocolNegotiator implements ProtocolNegotiator {
 
-    private DownstreamTlsContext downstreamTlsContext;
+    private final DownstreamTlsContext downstreamTlsContext;
     private final XdsClientWrapperForServerSds xdsClientWrapperForServerSds;
 
     ServerSdsProtocolNegotiator(
         DownstreamTlsContext downstreamTlsContext, int port, SynchronizationContext syncContext) {
-      this.downstreamTlsContext = downstreamTlsContext;
-      XdsClientWrapperForServerSds localXdsClientWrapperForServerSds;
+      this(downstreamTlsContext, getXdsClientWrapperForServerSds(port, syncContext));
+    }
+
+    private static XdsClientWrapperForServerSds getXdsClientWrapperForServerSds(
+        int port, SynchronizationContext syncContext) {
       try {
-        localXdsClientWrapperForServerSds =
-            XdsClientWrapperForServerSds.newInstance(port, Bootstrapper.getInstance(), syncContext);
+        return XdsClientWrapperForServerSds.newInstance(
+            port, Bootstrapper.getInstance(), syncContext);
       } catch (Exception e) {
-        logger.log(Level.WARNING, "Exception while creating the xDS client", e);
-        localXdsClientWrapperForServerSds = null;
+        logger.log(Level.FINE, "Fallback to plaintext due to exception", e);
```

Comment:
why we do this instead of using plaintext when creating the protocol negotiator?

Reply:
So for the logic, it does end up creating the plaintext protocol negotiator if you follow this flow: `null` for `XdsClientWrapperForServerSds` causes `SdsProtocolNegotiators.HandlerPickerHandler` on line 335 to replace with a plaintext negotiator. Is that what you were asking or something else?

Reply:
if we know it will be plaintext, the factory method `SdsProtocolNegotiators#serverProtocolNegotiator` can just return the simple plaintext PN not in the protocol negotiation handler if possible. we want to minimize conditional logic in each protocol to easy to maintain and debug. it can be very hard due to the async nature of handler.

Reply:
I think it can go both ways. For the `HandlerPickerHandler` you wanted (in a previous PR) proper handling of `ProtocolNegotiationEvent` to pick the actual Handler (plaintext or `ServerSdsHandler`)  based on `downstreamTlsContext` being empty. So that logic is centralized in one place even if the decision is delayed in the case you point out.

What you are saying can be done but will be a non-trivial change and I will have to create tests for it. Since this PR is really about server shutdown I am thinking of creating a TODO for me to do in another PR. What do you think?